homeassistant/components/coolmaster/climate.py

```diff
@@ -123,22 +122,30 @@ def name(self):
     @property
     def temperature_unit(self):
         """Return the unit of measurement."""
-        return self._unit
+        if self._unit.temperature_unit == "celsius":
```

Comment:
```suggestion
        if self.coolmaster_unit.temperature_unit == "celsius":
```

Reply:
`self._unit` is guaranteed to be up to date, but `self.coolmaster_unit` isn't.